packages/headless/src/lib/headless.service.ts

```diff
@@ -0,0 +1,531 @@
+import {
+  QueryClient,
+  QueryObserverOptions,
+  QueryObserverResult,
+  MutationObserverResult,
+} from '@tanstack/query-core';
+import io from 'socket.io-client';
+import { ApiService, IUserPreferenceSettings, IStoreQuery } from '@novu/client';
+import { IOrganizationEntity, IMessage } from '@novu/shared';
+
+import { QueryService } from './query.service';
+import type { ISession } from '../utils/types';
+import {
+  NOTIFICATIONS_QUERY_KEY,
+  ORGANIZATION_QUERY_KEY,
+  SESSION_QUERY_KEY,
+  UNSEEN_COUNT_QUERY_KEY,
+  USER_PREFERENCES_QUERY_KEY,
+} from '../utils';
+import {
+  FetchResult,
+  IHeadlessServiceOptions,
+  IUpdateActionVariables,
+  IUpdateUserPreferencesVariables,
+  UpdateResult,
+} from './types';
+
+export const NOTIFICATION_CENTER_TOKEN_KEY = 'nc_token';
+const isBrowser = typeof window !== 'undefined';
+const SESSION_NOT_INITIALIZED_ERROR =
+  'Session is not initialized, please use the initializeSession method first';
+
+const getToken = (): string | null => {
+  if (isBrowser) {
+    return localStorage.getItem(NOTIFICATION_CENTER_TOKEN_KEY);
+  }
+
+  return null;
+};
+
+export class HeadlessService {
```

Comment:
If we set an interface for `HeadlessService` would we avoid you have to set as `any` the instances in the tests?

Reply:
it wouldn't help, since the fields that I do check in the tests are private, but they are just private for the TS, not for the JS ;)